DayValue gives incorrect results when run in non-UTC environment

When running unit tests locally for `DayValue`, they fail and appear to be off by one day.

```
--- FAIL: TestDayValueYAML (0.00s)
    day_test.go:25: 
                Error Trace:    day_test.go:25
                Error:          Not equal: 
                                expected: []byte{0x64, 0x61, 0x79, 0x3a, 0x20, 0x22, 0x31, 0x39, 0x38, 0x35, 0x2d, 0x30, 0x36, 0x2d, 0x30, 0x32, 0x22, 0xa}
                                actual  : []byte{0x64, 0x61, 0x79, 0x3a, 0x20, 0x22, 0x31, 0x39, 0x38, 0x35, 0x2d, 0x30, 0x36, 0x2d, 0x30, 0x31, 0x22, 0xa}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,3 +1,3 @@
                                 ([]uint8) (len=18) {
                                - 00000000  64 61 79 3a 20 22 31 39  38 35 2d 30 36 2d 30 32  |day: "1985-06-02|
                                + 00000000  64 61 79 3a 20 22 31 39  38 35 2d 30 36 2d 30 31  |day: "1985-06-01|
                                  00000010  22 0a                                             |".|
                Test:           TestDayValueYAML
    day_test.go:47: 
                Error Trace:    day_test.go:47
                Error:          Not equal: 
                                expected: []byte{0x64, 0x61, 0x79, 0x3a, 0x20, 0x22, 0x31, 0x39, 0x38, 0x35, 0x2d, 0x30, 0x36, 0x2d, 0x30, 0x32, 0x22, 0xa}
                                actual  : []byte{0x64, 0x61, 0x79, 0x3a, 0x20, 0x22, 0x31, 0x39, 0x38, 0x35, 0x2d, 0x30, 0x36, 0x2d, 0x30, 0x31, 0x22, 0xa}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,3 +1,3 @@
                                 ([]uint8) (len=18) {
                                - 00000000  64 61 79 3a 20 22 31 39  38 35 2d 30 36 2d 30 32  |day: "1985-06-02|
                                + 00000000  64 61 79 3a 20 22 31 39  38 35 2d 30 36 2d 30 31  |day: "1985-06-01|
                                  00000010  22 0a                                             |".|
                Test:           TestDayValueYAML

```

Local machine:

```
$ timedatectl 
               Local time: Wed 2021-08-25 19:21:47 EDT
           Universal time: Wed 2021-08-25 23:21:47 UTC
                 RTC time: Wed 2021-08-25 23:21:47
                Time zone: America/New_York (EDT, -0400)
System clock synchronized: yes
              NTP service: active
          RTC in local TZ: no
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,6 @@
 ## Changelog
 
+* [CHANGE] Flagext: `DayValue` now always uses UTC when parsing or displaying dates. #71
 * [CHANGE] Closer: remove the closer package since it's trivial to just copy/paste. #70
 * [CHANGE] Memberlist: allow specifying address and port advertised to the memberlist cluster members by setting the following configuration: #37
   * `-memberlist.advertise_addr`

--- a/flagext/day.go
+++ b/flagext/day.go
@@ -25,12 +25,12 @@ func NewDayValue(t model.Time) DayValue {
 
 // String implements flag.Value
 func (v DayValue) String() string {
-	return v.Time.Time().Format(time.RFC3339)
+	return v.Time.Time().UTC().Format(time.RFC3339)
 }
 
 // Set implements flag.Value
 func (v *DayValue) Set(s string) error {
-	t, err := time.Parse("2006-01-02", s)
+	t, err := time.ParseInLocation("2006-01-02", s, time.UTC)
 	if err != nil {
 		return err
 	}
@@ -55,5 +55,5 @@ func (v *DayValue) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 // MarshalYAML implements yaml.Marshaler.
 func (v DayValue) MarshalYAML() (interface{}, error) {
-	return v.Time.Time().Format("2006-01-02"), nil
+	return v.Time.Time().UTC().Format("2006-01-02"), nil
 }

--- a/flagext/day_test.go
+++ b/flagext/day_test.go
@@ -9,8 +9,7 @@ import (
 )
 
 func TestDayValueYAML(t *testing.T) {
-	// Test embedding of DayValue.
-	{
+	t.Run("embedding DayValue", func(t *testing.T) {
 		type TestStruct struct {
 			Day DayValue `yaml:"day"`
 		}
@@ -28,10 +27,9 @@ func TestDayValueYAML(t *testing.T) {
 		err = yaml.Unmarshal(expected, &actualStruct)
 		require.NoError(t, err)
 		assert.Equal(t, testStruct, actualStruct)
-	}
+	})
 
-	// Test pointers of DayValue.
-	{
+	t.Run("pointer of DayValue", func(t *testing.T) {
 		type TestStruct struct {
 			Day *DayValue `yaml:"day"`
 		}
@@ -50,5 +48,5 @@ func TestDayValueYAML(t *testing.T) {
 		err = yaml.Unmarshal(expected, &actualStruct)
 		require.NoError(t, err)
 		assert.Equal(t, testStruct, actualStruct)
-	}
+	})
 }

